dns/r53/client.go

```diff
@@ -47,24 +52,35 @@ func (dns *client) GetHostedZoneDomain() (string, error) {
 }
 
 // UpdateRecordSets updates records in aws based on the change list.
-// Todo add tests
 func (dns *client) UpdateRecordSets(changes []*route53.Change) error {
-	recordSetsInput := &route53.ChangeResourceRecordSetsInput{
-		HostedZoneId: aws.String(dns.hostedZone),
-		ChangeBatch: &route53.ChangeBatch{
-			Changes: changes,
-		},
-	}
+	for i := 0; i < len(changes); i += dns.maxRecordChanges {
+		upperBound := min(i+dns.maxRecordChanges, len(changes))
+		batch := changes[i:upperBound]
+		log.Debug("Doing batch ", batch)
+		recordSetsInput := &route53.ChangeResourceRecordSetsInput{
+			HostedZoneId: aws.String(dns.hostedZone),
+			ChangeBatch: &route53.ChangeBatch{
+				Changes: batch,
+			},
+		}
 
-	_, err := dns.r53.ChangeResourceRecordSets(recordSetsInput)
+		_, err := dns.r53.ChangeResourceRecordSets(recordSetsInput)
 
-	if err != nil {
-		return fmt.Errorf("failed to create A record: %v", err)
+		if err != nil {
+			return fmt.Errorf("failed to create A record: %v", err)
+		}
```

Comment:
If only we had a generic partition function for slices :).

Do you think it would be clearer what's going on here if we created a partition method which breaks it up into batches first?